### PR TITLE
{cae}[intel/2015b] OpenFOAM 2.4.0 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
requires ~~https://github.com/hpcugent/easybuild-easyblocks/pull/688~~ and ~~https://github.com/hpcugent/easybuild-easyblocks/pull/692~~

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/c9ee577a42225f75f443 for a full test report.

**Comment by boegel** [comment]
@JensTimmerman: please review

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.4.0-intel-2015b.eb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+homepage = 'http://www.openfoam.com/'
+description = """OpenFOAM is a free, open source CFD software package. 
+ OpenFOAM has an extensive range of features to solve anything from complex fluid flows
+ involving chemical reactions, turbulence and heat transfer, 
+ to solid dynamics and electromagnetics."""
+
+toolchain = {'name': 'intel', 'version': '2015b'}
+
+source_urls = ['http://downloads.sourceforge.net/foam/%(version)s']
```
Comment (JensTimmerman) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.4.0-intel-2015b.eb]: I guess the sourceforge template doesn't work here?

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.4.0-intel-2015b.eb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+sources = [
+    SOURCE_TGZ,
+    'ThirdParty-%(version)s.tgz',
+]
+
+patches = [
+    'cleanup-OpenFOAM-%(version)s.patch',
+    'OpenFOAM-2.3.0_libreadline.patch',
```
Comment (JensTimmerman) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.4.0-intel-2015b.eb]: incorrect version?

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.4.0-intel-2015b.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.4.0-intel-2015b.eb]: nope, `foam` vs `OpenFOAM` (or `openfoam`, whatever)

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.4.0-intel-2015b.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.4.0-intel-2015b.eb]: the patch still works, no reason to create a new one

**Comment by JensTimmerman** [comment]
then feel free to merge

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/e44bbdfa100aba5c5385 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/94a95b09624d7451c371 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/9e7bf2717c869ff47899 for a full test report.

**Comment by JensTimmerman** [comment]
ok

**Comment by boegel** [comment]
@JensTimmerman: not OK: test report on CenOS7 failed (but probably because it was using the old OpenFOAM easyblock)

also, I just realised I forgot something in https://github.com/hpcugent/easybuild-easyblocks/pull/688...

will issue a follow-up PR for the OpenFOAM easyblock

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/192d2b765872d2b3a2ac for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2e2db16647b82223a031 for a full test report.

**Comment by boegel** [comment]
retested on top of https://github.com/hpcugent/easybuild-easyblocks/pull/692

works fine, but this has to wait until https://github.com/hpcugent/easybuild-easyblocks/pull/692 is merged (retesting of existing OpenFOAM builds is running)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/66f50afeb922477a17df for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/ee7cd08782c94c987121 for a full test report.

**Comment by boegel** [comment]
Good to go now https://github.com/hpcugent/easybuild-easyblocks/pull/688 and https://github.com/hpcugent/easybuild-easyblocks/pull/692 are merged in.

Thanks for the review @JensTimmerman!

**Pull request merged**